File: house-finder-be/HouseFinder360.Application/Common/Pagination/Pagination.cs
namespace HouseFinder360.Application.Common.Pagination;

public class Pagination
{
    private readonly int _currentPage = 1;
    private readonly int _pageSize = 10;
    public int CurrentPage 
    { 
        get => _currentPage;
        init => _currentPage = value > 0 ? value : 1; 
    }
    public int PageSize
    {
        get => _pageSize;
        init => _pageSize = value > 0 ? value : 10;
    }
    public int TotalItems { get; set; }
    public int TotalPages => (int)Math.Ceiling((decimal)TotalItems / PageSize);
}

File: house-finder-fe/src/core/interfaces/pagination.ts
export interface Pagination {
    currentPage: number,
    pageSize: number
}
export interface PaginationResponse<T> {
    pagination: PaginationInfo;
    data: T[]
}
export interface PaginationInfo {
    currentPage: number,
    pageSize : number,
    totalItems: number,
    totalPages: number
}